app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,24 @@
+class FriendshipsController < ApplicationController
+
+  def create
+    friend = User.by_email(friend_email)
+
+    if (friend.nil?)
+      flash[:error] = 'Your friend cannot be found. Are you sure they exist?'
+    else
+      friendship = Friendship.new(user_id: session[:user_id], friend_id: friend.id)
+
+      unless (friendship.save)
+        flash[:error] = 'Your friend could not be saved.'
+      end
+    end
+
+    redirect_to dashboard_path
+  end
+
+  private
+
+  def friend_email
+    params.permit(:friend_email)[:friend_email]
```

Comment:
curious about this syntax

Reply:
Sure!  What about it?

I wrote it like this b/c I only needed the actual email value to use in the controller action. The `params.permit` returns the hash-like params object (IDR the actual object type), and then I'm accessing the `friend_email` key to get the actual email value.

Reply:
awesome, that answers my question, i wasn't sure what the second :friend_email was doing